pyiceberg/io/pyarrow.py

```diff
@@ -387,15 +387,29 @@ def __init__(self, properties: Properties = EMPTY_DICT):
         super().__init__(properties=properties)
 
     @staticmethod
-    def parse_location(location: str) -> Tuple[str, str, str]:
-        """Return the path without the scheme."""
+    def parse_location(location: str, properties: Properties = EMPTY_DICT) -> Tuple[str, str, str]:
+        """Return (scheme, netloc, path) for the given location.
+
+        Uses DEFAULT_SCHEME and DEFAULT_NETLOC if scheme/netloc are missing.
+        """
         uri = urlparse(location)
+
+        default_scheme = properties.get("DEFAULT_SCHEME", "file")
+        default_netloc = properties.get("DEFAULT_NETLOC", "")
+
         if not uri.scheme:
-            return "file", uri.netloc, os.path.abspath(location)
-        elif uri.scheme in ("hdfs", "viewfs"):
-            return uri.scheme, uri.netloc, uri.path
+            scheme = default_scheme
+        if not uri.netloc:
+            netloc = default_netloc
+
+        if scheme in ("hdfs", "viewfs"):
+            return scheme, netloc, uri.path
         else:
-            return uri.scheme, uri.netloc, f"{uri.netloc}{uri.path}"
+            # For non-HDFS URIs, include netloc in the path if present
+            path = uri.path if uri.scheme else os.path.abspath(location)
+            if netloc and not path.startswith(netloc):
+                path = f"{netloc}{path}"
+            return scheme, netloc, path
```

Comment:
i dont remember the purpose of changing the logic inside the else block. i thought we only need to change when `uri.scheme` is empty

what do you think about this? 
```
        if not uri.scheme:
            # If no scheme is provided, default to file:// (local filesystem).
            # If properties contains DEFAULT_SCHEME, use that scheme instead of "file".
            default_scheme = properties.get("DEFAULT_SCHEME", "file")
            default_netloc = properties.get("DEFAULT_NETLOC", uri.netloc)
            return default_scheme, default_netloc, os.path.abspath(location)
        elif uri.scheme in ("hdfs", "viewfs"):
            return uri.scheme, uri.netloc, uri.path
        else:
            return uri.scheme, uri.netloc, f"{uri.netloc}{uri.path}"
```